src/main/java/com/revlearn/team1/service/transaction/TransactionServiceImp.java

```diff
@@ -1,56 +1,76 @@
+import com.stripe.param.PriceCreateParams;
+import com.stripe.param.ProductCreateParams;
+import com.stripe.param.checkout.SessionCreateParams;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
 import com.revlearn.team1.dto.transaction.TransactionRequestDTO;
-import com.revlearn.team1.dto.transaction.TransactionResponseDTO;
-import com.revlearn.team1.exceptions.TransactionNotFoundException;
 import com.revlearn.team1.mapper.TransactionMapper;
 import com.revlearn.team1.model.TransactionModel;
 import com.revlearn.team1.repository.TransactionRepo;
 
 @Service
 public class TransactionServiceImp implements TransactionService {
     @Autowired
-    private TransactionRepo transactionRepo;
+    TransactionRepo transactionRepo; // For future use with Kafka
 
     @Autowired
-    private TransactionMapper transactionMapper;
+    TransactionMapper transactionMapper;
 
-    @Override
-    public TransactionResponseDTO createTransaction(TransactionRequestDTO transaction) {
-        TransactionModel transactionModel = transactionMapper.fromDTO(transaction);
-        TransactionModel savedTransaction = transactionRepo.save(transactionModel);
-        return transactionMapper.toResDTO(savedTransaction);
-    }
+    @Value("${STRIPE_API_KEY}")
+    private String stripeApiKey;
 
-    @Override
-    public TransactionResponseDTO getTransactionById(int transactionId) {
-        TransactionModel retrievedTransaction = transactionRepo.findById(transactionId).orElseThrow(
-                () -> new TransactionNotFoundException(String
-                        .format("Could not find transaction by Id in Database.  Transaction ID: %d", transactionId)));
-        return transactionMapper.toResDTO(retrievedTransaction);
-    }
+    @Value("${CLIENT_URL}")
+    private String clientUrl;
 
     @Override
-    public List<TransactionResponseDTO> getTransactions() {
-        List<TransactionModel> transactionModels = transactionRepo.findAll();
-        return transactionModels.stream()
-                .map(transactionMapper::toResDTO)
-                .collect(Collectors.toList());
-    }
+    public TransactionResponseDTO checkout(TransactionRequestDTO request) throws StripeException {
+        TransactionModel item = transactionMapper.fromDTO(request);
 
-    @Override
-    public void deleteTransactionById(int transactionId) {
-        transactionRepo.findById(transactionId).orElseThrow(
-                () -> new TransactionNotFoundException(String.format(
-                        "Could not find transaction to delete by that Id in Database.  Transaction ID: %d",
-                        transactionId)));
+        // Set the Stripe API key
+        Stripe.apiKey = stripeApiKey;
+
+        // Create a Product in Stripe (optional, can be reused or stored in DB)
+        var productParams = ProductCreateParams.builder()
```

Comment:
We don't want to create a new product everytime there's a transaction, right? This apparently persists in Stripe. 

replace it with ProductData type which only creates Data for this individual transaction:
var productData = SessionCreateParams.LineItem.PriceData.ProductData.builder()
        .setName(item.getName()) // Set the product name or description
        .build();